doc/setup-binderhub.rst

```diff
@@ -148,6 +163,25 @@ Update ``config.yaml`` by entering the following::
      recommend something descriptive such as ``binder-dev-`` or ``binder-prod-``
      (ending with a `-` is useful).
 
+If you are using Azure Container Registry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want your BinderHub to push and pull images from an Azure Container Registry (ACR), then your `config.yaml` file will look as follows::
+
+    config:
+      BinderHub:
+        use_registry: true
+        image_prefix: <ACR_NAME>.azurecr.io/<project-name>/<prefix>-
+      DockerRegistry:
+        token_url: "https://<ACR_NAME>.azurecr.io/oauth2/token?service=<ACR_NAME>.azurecr.io"
+
+where:
+
+* `<ACR_NAME>` is the name you gave to your ACR,
+* `<project-name>` is an arbitrary name that is required due to BinderHub's hard dependency on the `gcr.io` image structure above.
```

Comment:
Can we replace "hard dependency" with "assumes that the `image_prefix` contains an extra level for the project name". We could fix this after all but so far haven't gotten round to it.

Do you think it would help people if we made a recommendation here as to what to use as a name? My recommendation would be to reuse `<ACR_NAME>`.

Reply:
Changing the language :+1:

It's tough to suggest a name as it has to be unique across _all_ of Azure. So I ended up with `testregistry1234sgibson91THISHADBETTERWORK` before my test registry was deployed into a unique namespace 🙂 This is why I suggested the `check-name` function in the setup registry docs.

Reply:
AH, I didn't mean to suggest a actual name name for ACR_NAME. I was thinking we should say that "a good value to use for `<project-name>` is the same value as you use for your `<ACR_PROJECT>`".

Reply:
Ah, I see! Yeah that would work. I've personally gone with the name I gave to the BinderHub, so if there's multiple Hubs attached, you can see what came from where.